From 643911ac0924dcb8671f641735c36738764154bc Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sat, 14 Jan 2023 07:07:53 +0200
Subject: [PATCH] [SW-118475] gaudi3: adapt ts counter for both dies

Timestamp counter code has a nice generic implementation where only
the base address of the block is required.
So far only Die0 psoc_timestamp counter was used. For the bootloader
app that will be added in the following patch, Die1 timestamp counter
will also be used by the instance that will execute on Die1 Arc1 CPU.
To Support that, the base address of the TS counter is provided
according to the location of the app execution.

Change-Id: I837fd8f57e5cd6d041c416ad072cd49de8d4a0e2
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 9f7f1ad..3bdd974 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -44,6 +44,7 @@
 #include <d1_pif_arc_regs.h>
 #include <d1_pif_arc_special_regs.h>
 #include <d1_psoc_boot_conf_regs.h>
+#include <d1_psoc_timestamp_regs.h>
 #include <gaudi3_blocks_linux_driver.h>
 
 #include <gaudi3/gaudi3_boot_info.h>
@@ -180,9 +181,19 @@
 	return ret;
 }
 
+u32 gaudi3_die_timestamp_base_get(u32 die_id)
+{
+	if (die_id == GAUDI3_SLAVE_DIE_ID)
+		return mmD1_PSOC_TIMESTAMP_CNTCR;
+
+	return mmD0_PSOC_TIMESTAMP_CNTCR;
+}
+
 u32 hl_pll_timestamp_base_get(void)
 {
-	return mmD0_PSOC_TIMESTAMP_CNTCR;
+	uint32_t die_id = hl_get_die_id();
+
+	return gaudi3_die_timestamp_base_get(die_id);
 }
 
 unsigned int get_cs_pad_sel_flash(void)
diff --git a/arch/gaudi3/arc/hl.h b/arch/gaudi3/arc/hl.h
index 85c5ef7..ae367a2 100644
--- a/arch/gaudi3/arc/hl.h
+++ b/arch/gaudi3/arc/hl.h
@@ -47,5 +47,6 @@
 bool hl_is_dual_die_sku(void);
 u64 gaudi3_dma_addr_fixup(u64 loadaddr);
 void pif_arc_pcie_monitoring(void);
+u32 gaudi3_die_timestamp_base_get(u32 die_id);
 
 #endif /* ARCH_GAUDI3_ARC_HL_H_ */
